dotCMS/src/main/java/com/dotcms/publishing/PublisherAPIImpl.java

```diff
@@ -86,9 +86,15 @@ public PublishStatus publish ( PublisherConfig config, PublishStatus status ) th
                 // If the bundle exists and we are retrying to push the bundle
                 // there is no need to run all the bundlers again.
                 if (!bundleExists || !publishAuditAPI.isPublishRetry(config.getId())) {
-                    PublishAuditHistory currentStatusHistory = publishAuditAPI
-                            .getPublishAuditStatus(config.getId()).getStatusPojo();
-                    currentStatusHistory.setBundleStart(new Date());
+                    PublishAuditStatus currentStatus = publishAuditAPI
+                            .getPublishAuditStatus(config.getId());
+                    PublishAuditHistory currentStatusHistory = null;
+                    if(currentStatus != null) {
+                        currentStatusHistory = currentStatus.getStatusPojo();
+                        if(currentStatusHistory != null) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)